pkg/route/api/validation/validation.go

```diff
@@ -32,6 +32,11 @@ func ValidateRoute(route *routeapi.Route) fielderrors.ValidationErrorList {
 		result = append(result, fielderrors.NewFieldInvalid("path", route.Spec.Path, "path must begin with /"))
 	}
 
+	if len(route.Spec.Path) > 0 && route.Spec.TLS != nil &&
+		route.Spec.TLS.Termination == routeapi.TLSTerminationPassthrough {
```

Comment:
Not saying we can't tighten this, I'm just saying we have to be very careful, and we need to audit how existing routes (especially route status) are being updated/validated:

> we can do it as long as we're sure that no automated systems will be hot looping or logging thousands of events/validation failures if a previously "valid" persisted object suddenly doesn't pass validation after an upgrade

We ran into this when validation around pod spec ports changed, and invalid replication controller data caused hundreds of thousands of pod creation failures and logged events.